btsieve/src/bitcoin/mod.rs

```diff
@@ -6,3 +6,161 @@ pub use self::{
     block_processor::{check_block_queries, check_transaction_queries},
     queries::{BlockQuery, TransactionQuery},
 };
+use crate::{AddBlock, Bitcoin};
+use bitcoin_support::{MinedBlock, Sha256dHash};
+
+impl AddBlock<MinedBlock> for Bitcoin {
+    fn add_block(&mut self, block: MinedBlock) {
+        if self.0.nodes.contains(&block) {
+            return log::warn!("Block already known {:?} ", block);
+        }
+        match self.find_predecessor(&block) {
+            Some(prev) => {
+                self.0.vertices.push((
+                    prev.block.header.merkle_root,
+                    block.clone().block.header.merkle_root,
+                ));
+            }
+            None => {
+                log::warn!("Could not find previous block for {:?} ", block);
+            }
+        }
+        self.0.nodes.push(block);
+    }
+
+    fn size(&self) -> usize {
+        self.0.nodes.len()
+    }
+
+    fn find_predecessor(&self, block: &MinedBlock) -> Option<&MinedBlock> {
+        self.0.nodes.iter().find(|b| {
+            b.block
+                .header
+                .merkle_root
+                .eq(&block.block.header.prev_blockhash)
+        })
+    }
+}
+
+#[cfg(test)]
+mod test {
+    use super::*;
+    use crate::BlockchainDAG;
```

Comment:
In line with Rust convention can we name this `BlockchainDag` please.  (Acronyms not capitalized, source: @luckysori)